akka-stream/src/main/scala/akka/stream/impl/fusing/Interpreter.scala

```diff
@@ -282,18 +319,48 @@ private[akka] class OneBoundedInterpreter(ops: Seq[Stage[_, _]], val forkLimit:
       null
     }
 
-    override def hold(): FreeDirective = {
-      if (currentOp.holding) throw new IllegalStateException("Cannot hold while already holding")
-      currentOp.holding = true
+    override def holdUpstream(): UpstreamDirective = {
+      removeBits(PrecedingWasPull)
+      addBits(UpstreamBall)
       exit()
     }
 
-    override def isHolding: Boolean = currentOp.holding
+    override def holdUpstreamAndPush(elem: Any): UpstreamDirective = {
+      ReactiveStreamsCompliance.requireNonNullElement(elem)
```

Comment:
Since fusing of operations is an optimization, we should not change semantics based on whether or not it has occurred: pushing an element downstream should be considered async / RS in all cases.